sonar-db/src/main/java/org/sonar/db/component/ComponentDto.java

```diff
@@ -160,7 +163,7 @@ public String name() {
   }
 
   public ComponentDto setName(String name) {
-    this.name = name;
+    this.name = checkComponentName(name);
```

Comment:
I think it's a new pattern, had a lot of discussions with @simonbrandhof 
Yes, I think it should be checked at dto level in the end to prevent any SQL error